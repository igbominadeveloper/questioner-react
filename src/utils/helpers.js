import jwt from 'jsonwebtoken';

export const decodeToken = token => {
  return jwt.decode(token);
};

export const setItem = (name, value) => {
  localStorage.setItem(name, value);
};

export const getItem = name => {
  const item = localStorage.getItem(name);
  return item;
};

export const destroyItem = item => {
  return localStorage.removeItem(item);
};

export const clearLocalStorage = () => {
  return localStorage.clear();
};
