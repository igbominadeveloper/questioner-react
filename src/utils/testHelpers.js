import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const setupStore = initialState => mockStore(initialState);

export const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};
