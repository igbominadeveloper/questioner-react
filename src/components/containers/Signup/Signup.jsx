import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

//modules
import { signupUser } from '../../../store/modules/auth';

//components
import Layout from '../../presentationals/Layout';

//stylesheets
import '../../../assets/css/Form.css';
import swal from 'sweetalert';
import { getItem } from '../../../utils/helpers';
import Loader from '../../presentationals/Loader/Loader';

export class Signup extends Component {
  state = {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    passwordConfirmation: '',
  };

  changeHandler = event => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  submitHandler = event => {
    event.preventDefault();
    const {
      password,
      passwordConfirmation,
      email,
      firstname,
      lastname,
    } = this.state;
    if (password !== passwordConfirmation) {
      swal('passwords do not match', '', 'error');
      return;
    }
    const redirectUrl = getItem('redirectUrl') || '/';
    this.props.signupUser(
      { email, password, firstname, lastname },
      redirectUrl,
      this.props.history,
    );
  };
  render() {
    return (
      <Layout>
        {this.props.isLoading && <Loader />}
        <div className="container">
          <div id="registration-form" className="shadow" data-test="reg-form">
            <form onSubmit={this.submitHandler} className="form">
              <p className="form-header">Welcome! Signup here</p>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control"
                  name="firstname"
                  value={this.state.firstname}
                  onChange={this.changeHandler}
                  id="firstname"
                  placeholder="Firstname"
                  required
                />
              </div>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control"
                  name="lastname"
                  value={this.state.lastname}
                  onChange={this.changeHandler}
                  id="lastname"
                  placeholder="Lastname"
                  required
                />
              </div>
              <div className="form-group">
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  value={this.state.email}
                  onChange={this.changeHandler}
                  id="email"
                  placeholder="Email Address"
                  required
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  value={this.state.password}
                  onChange={this.changeHandler}
                  id="password"
                  placeholder="Password"
                  required
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  name="passwordConfirmation"
                  value={this.state.passwordConfirmation}
                  onChange={this.changeHandler}
                  id="passwordConfirmation"
                  placeholder="Confirm Password"
                  required
                />
              </div>
              <div className="form-group">
                <button className="form-control btn-primary shadow">
                  Sign up
                </button>
              </div>
            </form>
            <p className="form-footer">
              <span>Have an account? </span>
              <Link to="/login" className="form-link text-primary">
                Sign in
              </Link>
            </p>
          </div>
        </div>
      </Layout>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  errors: state.auth.errors,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(
  mapStateToProps,
  { signupUser },
)(Signup);
