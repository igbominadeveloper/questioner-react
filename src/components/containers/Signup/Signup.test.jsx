import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from './Signup';

import { findByTestAttribute } from '../../../utils/testHelpers';

const component = shallow(<Signup />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the form section', () => {
    expect(findByTestAttribute(component, 'reg-form').length).toBe(1);
  });
});
