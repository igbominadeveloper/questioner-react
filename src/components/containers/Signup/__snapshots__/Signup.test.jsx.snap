// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<ConnectFunction>
  <div
    className="container"
  >
    <div
      className="shadow"
      data-test="reg-form"
      id="registration-form"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <p
          className="form-header"
        >
          Welcome! Signup here
        </p>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="firstname"
            name="firstname"
            onChange={[Function]}
            placeholder="Firstname"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="lastname"
            name="lastname"
            onChange={[Function]}
            placeholder="Lastname"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="passwordConfirmation"
            name="passwordConfirmation"
            onChange={[Function]}
            placeholder="Confirm Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="form-control btn-primary shadow"
          >
            Sign up
          </button>
        </div>
      </form>
      <p
        className="form-footer"
      >
        <span>
          Have an account? 
        </span>
        <Link
          className="form-link text-primary"
          to="/login"
        >
          Sign in
        </Link>
      </p>
    </div>
  </div>
</ConnectFunction>
`;
